thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -755,8 +764,13 @@ class ThinReplicaImpl {
           auto num_events = kvb_update.kv_pairs.size();
           auto client_id = getClientId(context);
           LOG_DEBUG(logger_, "Sending updates (history, data)" << KVLOG(client_id, block_id, num_events));
-
           sendData(stream, kvb_update);
+          metrics.last_sent_block_id.Get().Set(block_id);
+          last_sent_block_id_ = block_id;
+          if (++update_aggregator_counter == config_->update_metrics_aggregator_thresh) {
```

Comment:
no need to make a change on this but why not just use a module instead of ceqoing the counter?